src/Enterspeed.Source.UmbracoCms/Handlers/EnterspeedJobsHandler.cs

```diff
@@ -76,8 +77,25 @@ public void HandleJobs(IList<EnterspeedJob> jobs)
                 }
             }
 
+            var failedJobsToSave = new List<EnterspeedJob>();
+            foreach (var failedJob in failedJobs)
+            {
+                var existingJob = _enterspeedJobRepository.GetFailedJob(failedJob.EntityId);
+                if (existingJob != null)
+                {
+                    existingJob.Exception = failedJob.Exception;
+                    existingJob.CreatedAt = failedJob.CreatedAt;
+                    _enterspeedJobRepository.Update(existingJob);
+                }
+                else
+                {
+                    failedJobsToSave.Add(failedJob);
+                }
+            }
+
             // Save all jobs that failed
-            _enterspeedJobRepository.Save(failedJobs);
+            _enterspeedJobRepository.Save(failedJobsToSave);
```

Comment:
The HandleJobs method is getting quit big, I think it would makes sense to move this part to private method called SaveFailedJobs or something like that. 

Maybe it would also make sense to have the else part to simply save the job instead of building a list and the save afterwards, it would save a few lines of code and memory.